File: C/unit2/brute_force/linear_search.c
#include <stdio.h>

void linear(int a[], int n, int key)
{
    for (int i = 0; i < n; i++)
    {
        if (a[i] == key)
        {
            printf("%d found at index %d\n", key, i);
            return;
        }
    }
    printf("%d not found\n", key);
}

int main()
{
    int a[] = {1, 2, 3, 5, 9, 10, 14, 18};
    int n = 8;
    linear(a, n, 14);
}

File: Java/unit2/brute_force/linear_search.java
package Java.unit2.brute_force;

public class linear_search {
    static void linear(int a[], int n, int key) {
        for (int i = 0; i < n; i++) {
            if (a[i] == key) {
                System.out.println(key + " is found at index " + i);
                return;
            }
        }
        System.out.println("%d is not found");
    }

    public static void main(String[] args) {
        int a[] = { 1, 2, 3, 5, 9, 10, 14, 18 };
        int n = a.length;
        linear(a, n, 14);
    }
}
